utils/check_modular_conversion.py

```diff
@@ -120,55 +134,79 @@ def guaranteed_no_diff(modular_file_path, dependencies, models_in_diff):
     parser.add_argument(
         "--fix_and_overwrite", action="store_true", help="Overwrite the modeling_xxx.py file if differences are found."
     )
+    parser.add_argument("--check_all", action="store_true", help="Check all files, not just the ones in the diff.")
     parser.add_argument(
         "--num_workers",
-        default=1,
+        default=-1,
         type=int,
-        help="The number of workers to run. No effect if `fix_and_overwrite` is specified.",
+        help="The number of workers to run. Default is -1, which means the number of CPU cores.",
     )
     args = parser.parse_args()
     if args.files == ["all"]:
         args.files = glob.glob("src/transformers/models/**/modular_*.py", recursive=True)
 
+    if args.num_workers == -1:
+        args.num_workers = multiprocessing.cpu_count()
+
     # Assuming there is a topological sort on the dependency mapping: if the file being checked and its dependencies
     # are not in the diff, then there it is guaranteed to have no differences. If no models are in the diff, then this
     # script will do nothing.
     models_in_diff = get_models_in_diff()
-    if not models_in_diff:
+    if not models_in_diff and not args.check_all:
         console.print("[bold green]No models files or model tests in the diff, skipping modular checks[/bold green]")
         exit(0)
 
     skipped_models = set()
-    non_matching_files = 0
+    non_matching_files = []
     ordered_files, dependencies = find_priority_list(args.files)
-    if args.fix_and_overwrite or args.num_workers == 1:
-        for modular_file_path in ordered_files:
-            is_guaranteed_no_diff = guaranteed_no_diff(modular_file_path, dependencies, models_in_diff)
-            if is_guaranteed_no_diff:
-                model_name = modular_file_path.rsplit("modular_", 1)[1].replace(".py", "")
-                skipped_models.add(model_name)
+    flat_ordered_files = [item for sublist in ordered_files for item in sublist]
+
+    console.print(f"[bold yellow]Number of dependency levels: {len(ordered_files)}[/bold yellow]")
+    console.print(f"[bold yellow]Files per level: {tuple([len(x) for x in ordered_files])}[/bold yellow]")
+
+    try:
+        for dependency_level_files in ordered_files:
+            # Filter files guaranteed no diff
+            files_to_check = []
+            for file_path in dependency_level_files:
+                if not args.check_all and guaranteed_no_diff(file_path, dependencies, models_in_diff):
+                    skipped_models.add(file_path.split("/")[-2])  # save model folder name
+                else:
+                    files_to_check.append(file_path)
```

Comment:
so `files_to_check` a list of of modular file paths and there won't be any duplicated elements?